sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -417,6 +417,57 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
           Context const& context) const override;
     };
 
+    /**
+     * @brief Helper class for creating scopes for the token request context.
+     *
+     */
+    struct TokenScopes
+    {
+      /**
+       * @brief Parses url and generates a scope from it.
+       *
+       * @details For example, from a url like: `https://<account>.<service>.azure.net` the
+       * generated scope is in the form of: `https://<service>.azure.net/.default`.
+       *
+       * @remark As per Azure rules, the account for any service  name must only contain
+       * alphanumeric characters and dashes and cannot start with a number. This allow this method
+       * to calculate the scope after the first `.` from the url host.
+       *
+       * @param url The url used to create an Azure SDK client.
+       * @param defaultScope Optional hardcoded scope to be always added to the result scopes. It
+       * won't be added if it is equal to the calculated scope from the url.
+       * @return A vector with the calculated scopes plus any default scope provided.
+       */
+      static std::vector<std::string> GetScopeFromUrl(
+          Azure::Core::Url const& url,
+          std::string const& defaultScope = "")
+      {
+        std::vector<std::string> scopes;
+
+        std::string calculatedScope(url.GetScheme() + "://");
+        auto const& hostWithAccount = url.GetHost();
+        auto hostNoAccountStart = std::find(hostWithAccount.begin(), hostWithAccount.end(), '.');
```

Comment:
Let's make it a non-inline function, so that a header does not have to include `<algorithm>`, `<vector>` and so on.

Reply:
sure

Reply:
o wait, algorithm and vector where there even before this function